pkg/upload/dsym.go

```diff
@@ -36,82 +38,105 @@ func ProcessDsym(
 ) error {
 
 	var buildSettings *ios.XcodeBuildSettings
-	var dsyms *[]*ios.DwarfInfo
 	var plistData *ios.PlistData
 	var uploadOptions map[string]string
 
-	for _, path := range paths {
-		uploadInfo, err := ios.ProcessPathValues(path, dsymPath, projectRoot)
-		if err != nil {
-			return err
+	var dwarfInfo []*ios.DwarfInfo
+	var tempDirs []string
+
+	// Performs an automatic cleanup of temporary directories at the end
+	defer func() {
+		for _, tempDir := range tempDirs {
+			_ = os.RemoveAll(tempDir)
 		}
+	}()
+
+	for _, path := range paths {
 
-		// If projectRoot is not set explicitly, use uploadInfo to set it (where possible)
-		if projectRoot == "" && uploadInfo.ProjectRoot != "" {
-			projectRoot = uploadInfo.ProjectRoot
+		if path == "" {
+			// set path to current directory if not set
+			path, _ = os.Getwd()
 		}
 
-		// If scheme is set explicitly, check if it exists
-		if scheme != "" && len(uploadInfo.DsymPaths) == 0 {
-			_, err = ios.IsSchemeInPath(path, scheme)
-			if err != nil {
-				return err
-			}
-			log.Info("Using scheme: " + scheme)
-		} else {
+		if ios.IsPathAnXcodeProjectOrWorkspace(path) {
+			projectRoot = ios.GetDefaultProjectRoot(path, projectRoot)
+			log.Info("Defaulting to '" + projectRoot + "' as the project root")
+
+			// Get build settings and dsymPath
+
+			// If scheme is set explicitly, check if it exists
+			if scheme != "" {
+				_, err := ios.IsSchemeInPath(path, scheme)
+				if err != nil {
+					log.Warn(err.Error())
+				}
 
-			// Only when the dsym path is not set, try and work out the scheme
-			if dsymPath == "" && len(uploadInfo.DsymPaths) == 0 {
-				// If the scheme is not set explicitly, try to find it
+			} else {
+				// Otherwise, try to find it
+				var err error
 				scheme, err = ios.GetDefaultScheme(path)
 				if err != nil {
-					return err
+					log.Warn(err.Error())
 				}
-				log.Info("Using scheme: " + scheme)
+
 			}
-		}
 
-		// If the dsymPath is not fed in via <path>
-		if dsymPath == "" && len(uploadInfo.DsymPaths) == 0 {
+			var err error
 			buildSettings, err = ios.GetXcodeBuildSettings(path, scheme)
```

Comment:
scheme can be nil here - if we didn't find it in the default. So this should be conditioned and on L90 it should check `buildSettings` being nil.